From d4614eaeb3d5eceb486320af3f9b19790a6cfd0d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 08 Nov 2021 18:35:24 +0200
Subject: [PATCH] [SW-0] gaudi2: pll: obey postdiv1 >= postdiv2

Reorder the postdiv values as the PLL datasheet prescribes.
The datasheet defines POSTDIV1 >= POSTDIV2. For this to hold, swap the
incorrectly places values.
The resulting frequency is not changed.

Change-Id: I172cc530ea56630bec58971ec64f55364913ef01
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index eccb5af..4f3061c 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -35,7 +35,7 @@
 	{
 		.name = "DCORE0_IF",
 		.base = mmDCORE0_XBAR_IF_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 1, .nb = 2,	/* 1600 */
+		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
 		.div_factor = { 1, 2, 3, 7 },
 		.div_sel = { HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
@@ -45,7 +45,7 @@
 	}, {
 		.name = "DCORE1_IF",
 		.base = mmDCORE1_XBAR_IF_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 1, .nb = 2,	/* 1600 */
+		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
 		.div_factor = { 1, 2, 3, 7 },
 		.div_sel = { HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
@@ -55,7 +55,7 @@
 	}, {
 		.name = "DCORE2_IF",
 		.base = mmDCORE2_XBAR_IF_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 1, .nb = 2,	/* 1600 */
+		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
 		.div_factor = { 1, 2, 3, 7 },
 		.div_sel = { HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
@@ -65,7 +65,7 @@
 	}, {
 		.name = "DCORE3_IF",
 		.base = mmDCORE3_XBAR_IF_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 1, .nb = 2,	/* 1600 */
+		.nr = 1, .nf = 64, .od = 2, .nb = 1,	/* 1600 */
 		.div_factor = { 1, 2, 3, 7 },
 		.div_sel = { HL_PLL_DIV_PLL,
 			     HL_PLL_DIV_PLL,
@@ -95,7 +95,7 @@
 	}, {
 		.name = "DCORE2_SRAM",
 		.base = mmDCORE2_XBAR_BANK_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 2, .nb = 4,	/* 400 */
+		.nr = 1, .nf = 64, .od = 4, .nb = 2,	/* 400 */
 		.div_factor = { 0, -1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
@@ -105,7 +105,7 @@
 	}, {
 		.name = "DCORE3_SRAM",
 		.base = mmDCORE3_XBAR_BANK_PLL_CTRL_BASE,
-		.nr = 1, .nf = 64, .od = 2, .nb = 4,	/* 400 */
+		.nr = 1, .nf = 64, .od = 4, .nb = 2,	/* 400 */
 		.div_factor = { 0, -1, -1, -1 },
 		.div_sel = { HL_PLL_PLL_CLK,
 			     HL_PLL_REF_CLK,
